lib/int64.js

```diff
@@ -0,0 +1,373 @@
+'use strict';
+
+const { Uint64 } = require('./uint64');
+
+const INT32_MAX = 0x7FFFFFFF;
+const UINT32_MAX = 0xFFFFFFFF;
+
+const charToNum = {
+  0: 0, 1: 1, 2: 2,
+  3: 3, 4: 4, 5: 5,
+  6: 6, 7: 7, 8: 8,
+  9: 9,
+  a: 10, A: 10,
+  b: 11, B: 11,
+  c: 12, C: 12,
+  d: 13, D: 13,
+  e: 14, E: 14,
+  f: 15, F: 15,
+};
+
+class Int64 {
+  constructor(value) {
+    this.value = new Uint32Array(2);
+    if (value instanceof Int64) {
+      this.value[0] = value.value[0];
+      this.value[1] = value.value[1];
+      return;
+    }
+    const numValue = Number(value);
+    if (Number.isNaN(numValue)) return;
+    if (numValue >= 0 && numValue <= INT32_MAX) {
+      this.value[0] = numValue;
+    } else if (numValue >= -INT32_MAX && numValue < 0) {
+      this.value[0] = numValue;
+      this.value[1] = UINT32_MAX;
+    } else if (numValue >= 0 && numValue <= Number.MAX_SAFE_INTEGER) {
+      this.value[0] = numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(numValue / (UINT32_MAX + 1));
+    } else if (numValue >= Number.MIN_SAFE_INTEGER && numValue < 0) {
+      this.value[0] = -numValue % (UINT32_MAX + 1);
+      this.value[1] = Math.floor(-numValue / (UINT32_MAX + 1));
+      this.value = Int64._conversion(this.value);
+    } else if (typeof value === 'string') {
+      let sign = false;
```

Comment:
I'd prefer to have more explicit name, i.e. `negative`/`neg`?

Reply:
`negative === true` if value `< 0`
I'd propose `positive === true` if value `> 0`